src/Sentry.Maui/SessionReplay.cs

```diff
@@ -0,0 +1,63 @@
+#if __ANDROID__
+using View = Android.Views.View;
+using Android.Views;
+using Java.Lang;
+using Microsoft.Maui.Handlers;
+using Microsoft.Maui.Platform;
+#endif
+
+namespace Sentry.Maui;
+
+/// <summary>
+/// Contains custom <see cref="BindableProperty"/> definitions used to control the behaviour of the Sentry SessionReplay
+/// feature in MAUI apps.
+/// </summary>
+public static class SessionReplay
+{
+    /// <summary>
+    /// Mask can be used to either unmask or mask a view.
+    /// </summary>
+    public static readonly BindableProperty MaskProperty =
+        BindableProperty.CreateAttached(
+            "Mask",
+            typeof(SessionReplayMaskMode),
+            typeof(SessionReplay),
+            defaultValue: SessionReplayMaskMode.Mask,
+            propertyChanged: OnMaskChanged);
+
+    /// <summary>
+    /// Gets the value of the Mask property for a view
+    /// </summary>
+    public static SessionReplayMaskMode GetMask(BindableObject view) => (SessionReplayMaskMode)view.GetValue(MaskProperty);
+
+    /// <summary>
+    /// Sets the value of the Mask property for a view. .
+    /// </summary>
+    /// <param name="view">The view element to mask or unmask</param>
+    /// <param name="value">The value to assign. Can be either "sentry-mask" or "sentry-unmask".</param>
+    public static void SetMask(BindableObject view, SessionReplayMaskMode value) => view.SetValue(MaskProperty, value);
+
+    private static void OnMaskChanged(BindableObject bindable, object oldValue, object newValue)
+    {
+#if __ANDROID__
+            if (bindable is VisualElement ve)
+            {
+                ve.HandlerChanged += (s, e) =>
+                {
+                    if (ve.Handler?.PlatformView is not View nativeView ||
+                        newValue is not SessionReplayMaskMode maskSetting)
+                    {
+                        return;
+                    }
+
+                    nativeView.Tag = maskSetting switch
+                    {
+                        SessionReplayMaskMode.Mask => "sentry-mask",
```

Comment:
question: Is it worth consolidating both mask-strings to internal consts? (e.g. within `internal static class SentryTags` type)

Reply:
We could. This maps to a detail of the implementation of this feature in the Java/Android SDK... I imagine it will be different for iOS. Currently it's only used in two places - here and in the [MauiEventsBinder](https://github.com/getsentry/sentry-dotnet/blob/d51862043a7684d8fa6cbb596674ee73a253c891/src/Sentry.Maui/Internal/MauiEventsBinder.cs#L442).

I've added an extension method (conditionally compiled for Android) to map from the "strongly typed" enum we have in .NET to the mask strings used by the Java SDK. That centralises and documents the magic strings, so is hopefully a better solution.